port/common/omrsysinfo.c

```diff
@@ -808,42 +808,73 @@ omrsysinfo_os_kernel_info(struct OMRPortLibrary *portLibrary, struct OMROSKernel
 }
 
 /**
- * Checks if the port library can use cgroup limits
+ * Checks if the cgroup system is available
  *
  * @param[in] portLibrary pointer to OMRPortLibrary
  *
- * @return 0 if the port library can use cgroup limits, otherwise negative error code
+ * @return 0 if the the cgroup system is available, otherwise negative error code
  */
-int32_t 
-omrsysinfo_cgroup_is_limits_supported(struct OMRPortLibrary *portLibrary)
+int32_t
+omrsysinfo_cgroup_system_is_available(struct OMRPortLibrary *portLibrary)
 {
-	return OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
+	return OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;;
 }
 
 /**
- * Returns TRUE if port library is using cgroup limits, FALSE otherwise
+ * Returns cgroup subsystems available for port library to use
  *
  * @param[in] portLibrary pointer to OMRPortLibrary
  *
- * @return TRUE if port library has been enabled to use cgroup limits, FALSE otherwise
+ * @return bitwise-OR of flags of type OMR_CGROUP_SUBSYSTEMS_* indicating the
+ * subsystems that are available
  */
-BOOLEAN 
-omrsysinfo_cgroup_is_limits_enabled(struct OMRPortLibrary *portLibrary)
+uint64_t
+omrsysinfo_cgroup_get_available_subsystems(struct OMRPortLibrary *portLibrary)
+{
+	return 0;
+}
+
+/**
+ * Returns TRUE if specified cgroup subsystem is available for port library to use, FALSE otherwise
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[in] subsystem one of the flags of type OMR_CGROUP_SUBSYSTEMS_*
+ *
+ * @return TRUE if specified cgroup subsystem is available for port library to use, FALSE otherwise
+ */
+BOOLEAN
+omrsysinfo_cgroup_is_subsystem_available(struct OMRPortLibrary *portLibrary, uint64_t subsystem)
 {
 	return FALSE;
 }
 
 /**
- * Enable port library to use cgroup limits 
+ * Returns cgroup subsystems enabled for port library to use
  *
- * @param[in] portLibrary pointer to OMRPortLibrary 
+ * @param[in] portLibrary pointer to OMRPortLibrary
  *
- * @return 0 if the port library can use cgroup limits, otherwise negative error code
+ * @return bitwise-OR of flags of type OMR_CGROUP_SUBSYSTEMS_* indicating the
+ * subsystems that are enbaled
  */
-int32_t 
-omrsysinfo_cgroup_enable_limits(struct OMRPortLibrary *portLibrary)
+uint64_t
+omrsysinfo_cgroup_get_enabled_subsystems(struct OMRPortLibrary *portLibrary)
 {
-	return OMRPORT_ERROR_SYSINFO_CGROUP_UNSUPPORTED_PLATFORM;
+	return 0;
+}
+
+/**
+ * Enable port library to use specified cgroup subsystems
+ *
+ * @param[in] portLibrary pointer to OMRPortLibrary
+ * @param[in] subsystems bitwise-OR of flags of type OMR_CGROUP_SUBSYSTEMS_* 
+ * indicating the subsystems to be enabled 
+ *                           
+ * @return bitwise-OR of flags of type OMR_CGROUP_SUBSYSTEMS_* indicating the subsystems that are enabled
+ */
+uint64_t
+omrsysinfo_cgroup_enable_subsystems(struct OMRPortLibrary *portLibrary, uint64_t requestedSubsystems)
+{
```

Comment:
How are errors handled / communicated to the caller by this function?

What if the user tries to enable a subsystem that isn't enabled in the system?

Reply:
Actually there is no need to call omrsysinfo_cgroup_get_available_subsystems() here:

```
if (0 == omrsysinfo_cgroup_is_system_available()) {
    uint64_t enabledSubsystems = omrsysinfo_cgroup_enable_subsystems(requiredSubsystems);
    if ((enabledSubsystems & requiredSybsystems) != enabledSubsystems) {
            /* do error handling */
    }
}
```